napari/layers/labels/_labels_utils.py

```diff
@@ -201,3 +203,18 @@ def mouse_event_to_labels_coordinate(layer, event):
             return None
         coordinates = first_nonzero_coordinate(layer.data, start, end)
     return coordinates
+
+
+def mouse_event_to_nd_line_plane_intersection(layer, event):
+    painting_plane = layer.experimental_linked_image_layer.plane
```

Comment:
You might be able to make this slightly more general. All you need here is something that provides the `painting_plane` (which is just something that provides `intersect_with_line`).

I could see how it might be natural for a label layer (or at least the painting part of it) to track an object that provides that `intersect_with_line` capability - it doesn't have to be a plane.

Reply:
Love this! That sounds like a nice general interface we could adhere to - I'm excited that you find it natural